src/contract_error.ts

```diff
@@ -0,0 +1,77 @@
+import { xdr } from 'stellar-sdk';
+
+export class ContractError extends Error {
```

Comment:
thoughts on including a "sentToNetwork" boolean on this object?

Consumers likely would find value in knowing if this error was caught in simulation or caught on submission.

However, this probably won't matter after the refactor (assuming running simulation is a distinct step for the consumer), so we can probably push this?

Reply:
This should be taken care of in the refactor